hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/connector/HazelcastWriters.java

```diff
@@ -487,4 +483,64 @@ private boolean isRemote() {
                          .limit(count).collect(toList());
         }
     }
+
+    public static class ApplyFnEntryProcessor<K, V, T> implements EntryProcessor<K, V>, EntryBackupProcessor,
+            IdentifiedDataSerializable {
+        private Map<K, T> tmpMap;
+        private DistributedBiFunction<V, T, V> updateFn;
+
+        public ApplyFnEntryProcessor() {
+        }
+
+        public ApplyFnEntryProcessor(Map<K, T> tmpMap, DistributedBiFunction<V, T, V> updateFn) {
+            this.tmpMap = tmpMap;
+            this.updateFn = updateFn;
+        }
+
+        @Override
+        public Object process(Entry<K, V> entry) {
+            V oldValue = entry.getValue();
+            T item = tmpMap.get(entry.getKey());
+            V newValue = updateFn.apply(oldValue, item);
+            entry.setValue(newValue);
+            return null;
+        }
+
+        @Override public EntryBackupProcessor<K, V> getBackupProcessor() {
+            return this;
+        }
+
+        @Override public void writeData(ObjectDataOutput out) throws IOException {
+            out.writeInt(tmpMap.size());
+            for (Entry<K, T> entry : tmpMap.entrySet()) {
```

Comment:
there's already serializer for HashMap (see HashMapHook), so you don't need to re-implement it here.